plugin/arcconf/arcconf.py

```diff
@@ -644,3 +644,98 @@ def volume_delete(self, volume, flags=0):
+
+        volume_info = volume.plugin_data
+        ctrl_id = str(volume_info['ctrl_id'])
+        array_id = str(volume_info['array_id'])
+        stripe_size = int(volume_info['stripe_size']) * 1024  # convert to Kibibyte
+        full_stripe_size = int(volume_info['full_stripe_size']) * 1024  # convert to Kibibyte
+        device_count = 0
+
+        array_info = self._arcconf_exec(['GETCONFIGJSON', ctrl_id, 'ARRAY', array_id], flag_force=True)
+        array_json_info = self._filter_cmd_output(array_info)['Array']
+        for chunk in array_json_info['Chunk']:
+            if 'deviceID' in chunk.keys():
+                device_count += 1
+            else:
+                continue
+
+        raid_level = _ARCCONF_RAID_LEVEL_CONV[str(volume_info['raid_level'])]
+
+        if device_count == 0:
+            if stripe_size == Volume.STRIP_SIZE_UNKNOWN:
+                raise LsmError(
+                    ErrorNumber.PLUGIN_BUG,
+                    "volume_raid_info(): Got logical drive %s entry, " %
+                    volume.id + "but no physicaldrive entry")
+
+            raise LsmError(
+                ErrorNumber.NOT_FOUND_VOLUME,
+                "Volume not found")
+
+        return [raid_level, stripe_size, device_count, stripe_size, full_stripe_size]
+
+    @_handle_errors
+    def pool_member_info(self, pool, flags=Client.FLAG_RSVD):
+        """
+
+        :param pool: pool id -  Pool to query
+        :param flags: Optional
+        :return: [raid_type, member_type, member_ids]
+
+        Depends on command:
+            arcconf getconfigjson <ctrlNo>
+        """
+
+        if not pool.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input pool argument: missing plugin_data property")
+
+        pool_info = pool.plugin_data
+        ctrl_id = pool_info['ctrl_id']
+        array_id = pool_info['array_id']
+        consumer_array_id = None
+        raid_level = Volume.RAID_TYPE_UNKNOWN
+        device_id = []
+
+        ctrl_info = self._arcconf_exec(['GETCONFIGJSON', ctrl_id])
+        ctrl_json_info = self._filter_cmd_output(ctrl_info)
+        device_info = ctrl_json_info['Controller']['Channel']
+        volume_info = ctrl_json_info['Controller']['LogicalDrive']
+
+        for volume in volume_info:
+            chunk_data = volume['Chunk']
+            for chunk in chunk_data:
+                consumer_array_id = chunk['consumerArrayID']  # consumerArrayID in all the chunk will be same
+            if consumer_array_id == array_id:
+                raid_level = volume['raidLevel']
+
+        for device in device_info:
+            if 'HardDrive'in device.keys():
+                for hard_drive in device['HardDrive']:
+                    chunk_data = hard_drive['Chunk']
+                    for chunk in chunk_data:
+                        if ('consumerArrayID' in chunk.keys()) and (chunk['consumerArrayID'] == array_id):
```

Comment:
The text width should be less than 80.

Reply:
done - commit id - 9a67e0bbaf53081e2da601513d85b0d33b767ba8